public/controller/data_functions.php

```diff
@@ -278,8 +278,11 @@ function upload_homework($username, $semester, $course, $assignment_id, $homewor
 
     $settings_file = $user_path."/user_assignment_settings.json";
     if (!file_exists($settings_file)) {
-        $json = array("active_assignment"=>$upload_version);
-        file_put_contents($settings_file, json_encode($json));
+        date_default_timezone_set('America/New_York');
```

Comment:
Would setting the default timezone in index.php under umask set everything in the data_functions.php?

The timezone settings is set in other places as well:

```
$ git grep date_default
TAGradingServer/lib/Logger.php:        date_default_timezone_set("America/New_York");
TAGradingServer/toolbox/functions.php:date_default_timezone_set('America/New_York');
public/index.php:date_default_timezone_set('America/New_York');
tests/unitTests/lib/ExceptionHandlerTester.php:    date_default_timezone_set("America/New_York");
```